charts/kube-prometheus-stack/templates/prometheus/rules-1.14/alertmanager.rules.yaml

```
@@ -42,12 +42,12 @@ spec:
         # Without max_over_time, failed scrapes could create false negatives, see
         # https://www.robustperception.io/alerting-on-gauges-in-prometheus-2-0 for details.
         max_over_time(alertmanager_config_last_reload_successful{job="{{ $alertmanagerJob }}",namespace="{{ $namespace }}"}[5m]) == 0
-      for: 10m
+      for: {{ dig "AlertmanagerFailedReload" "for" "10m" .Values.customRules }}
       {{- with .Values.defaultRules.keepFiringFor }}
       keep_firing_for: "{{ . }}"
       {{- end }}
       labels:
-        severity: critical
+        severity: {{ dig "AlertmanagerFailedReload" "labelsSeverity" "critical" .Values.customRules }}
```

is there a reason for dig instead

```suggestion
        severity: {{ .Values.customRules.AlertmanagerFailedReload.severity | default "critical" }}
```

I would prefer `severity` instead `labelsSeverity`

this pattern is more commonly used.

---

I tried different approaches but preferred dig, as it does not require to set the keys in the values yaml. So we don't end up with an entry for each alert rule like
```
customRules:
  AlertmanagerFailedReload: {}
  AlertmanagerMembersInconsistent: {}
  ...
```
With not set keys a nil pointer error is thrown.
```
Error: template: kube-prometheus-stack/templates/prometheus/rules-1.14/alertmanager.rules.yaml:45:21: executing "kube-prometheus-stack/templates/prometheus/rules-1.14/alertmanager.rules.yaml" at <.V
alues.customRules.AlertmanagerFailedReload.for>: nil pointer evaluating interface {}.for
```

Means each level of the keys had to be checked if it exists, which makes the template harder to read / maintain.
"dig" solves that for you :)

I am happy to change it from labelsSeverity to severity. My reason for labelsSeverity was the helm approach on [flat/nested values](https://helm.sh/docs/chart_best_practices/values/#flat-or-nested-values) as it shows the levels a bit nicer like lables.Severity in the manifests, and you can derive the hierachy. But if you still prefer serverity I change it.

---

@jkroepke i've update the PR and used 'severity' instead of 'labelsSeverity' , as it keeps the sync script a bit easier.